File: src/main/java/cn/labzen/logger/spring/Profiles.java
package cn.labzen.logger.spring;

import java.util.List;

public final class Profiles {

  public static final String DEFAULT = "";
  public static final String DEV = "dev";
  public static final String PROD = "prod";
  public static final String TEST = "test";
  public static final String UAT = "uat";
  public static final String DEV_FULL = "development";
  public static final String PROD_FULL = "production";

  private static volatile List<String> currentProfiles = List.of();

  private Profiles() {
  }

  public static List<String> currentProfiles() {
    return currentProfiles;
  }

  static void setCurrentProfiles(List<String> profiles) {
    currentProfiles = List.copyOf(profiles);
  }
}

File: src/main/kotlin/cn/labzen/logger/spring/Profiles.kt
package cn.labzen.logger.spring

object Profiles {

  const val DEFAULT: String = ""
  const val DEV: String = "dev"
  const val DEVELOPMENT: String = "development"
  const val TEST: String = "test"
  const val PROD: String = "prod"
  const val PRODUCTION: String = "production"

  internal lateinit var currentProfiles: List<String>

  // fun setCurrentProfiles(profiles: Array<String>) {
  //   currentProfiles = profiles.toList()
  // }
}
